### DISCUSSION TITLE
Idiomatic way to implement a heartbeat

### DISCUSSION THREAD

**Discussion by riverar** [open-telemetry/community#1598] [Community Discussions]
Is there a documented pattern or generally accepted method of implementing a _heartbeat_ in a desktop application? That is, a periodic message send to a collector for the purpose of determining an active session count? Or perhaps a general pattern of reporting back Boolean states?

I was going down the path of using a Counter with a 0|1 value and just letting that contact a Collector periodically, but wanted to double-check.

My notes thus far:
* Metrics reader can be configured non-cumulative for fire once
   https://github.com/open-telemetry/opentelemetry-dotnet/issues/2524

* On-going work related to establishing conventions for service uptime
   https://github.com/open-telemetry/oteps/pull/185

**Comment by trask** [comment]
Hi @riverar! I think the best place to ask this question is probably in https://github.com/open-telemetry/semantic-conventions

**Reply by riverar** [reply]
Thanks @trask, done. https://github.com/open-telemetry/semantic-conventions/issues/199
